### PR TITLE
Pagy 8 support

### PR DESCRIPTION
**Pull request by jasl**
This PR mixed a few of things:
- Loosen requirement to support Pagy 8
- Fix typo
- Slightly restructure file organization
- bump version
- Add Ruby 3.3 to the CI matrix

### CODE REVIEW

File: pagy_cursor.gemspec [hunk 1]
```diff
@@ -22,10 +22,10 @@ Gem::Specification.new do |spec|
   spec.bindir        = "exe"
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
-  spec.add_dependency "pagy", ">= 6", "< 7"
+  spec.add_dependency "pagy", ">= 6"
```
Comment (Uysim) [inline pagy_cursor.gemspec:25]: If you test with pagy 8. Please specific 
```
spec.add_dependency "pagy", ">= 6", "< 9"
```

File: lib/pagy_cursor/version.rb [hunk 2]
```diff
@@ -1,3 +0,0 @@
-module PagyCursor
```
Comment (Uysim) [inline lib/pagy_cursor/version.rb:1]: Why are we removing this file?

Comment (jasl) [reply lib/pagy_cursor/version.rb:1]: Migrate this constant to `pagy_cursor.rb` for simplicity.
I can revert the change.

The PR was my personal hack originally, so combine a few of off-topic changes

File: pagy_cursor.gemspec (hunk 1 above)
Comment (jasl) [reply pagy_cursor.gemspec]: fixed

**Comment by jasl** [comment]
BTW, this branch works for my projects, but I don't know why `rake spec` has problem

```
/Users/jasl/.rbenv/versions/3.3.0/bin/ruby -I/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-support-3.13.1/lib:/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/lib /Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb

An error occurred while loading spec_helper.
Failure/Error: require "rails/test_help"

NoMethodError:
  undefined method `config' for nil
# ./spec/spec_helper.rb:4:in `<top (required)>'
No examples found.


Finished in 0.00002 seconds (files took 0.63226 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples

/Users/jasl/.rbenv/versions/3.3.0/bin/ruby -I/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-support-3.13.1/lib:/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/lib /Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb failed
```

**Comment by Uysim** [comment]
> BTW, this branch works for my projects, but I don't know why `rake spec` has problem
> 
> ```
> /Users/jasl/.rbenv/versions/3.3.0/bin/ruby -I/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-support-3.13.1/lib:/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/lib /Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
> 
> An error occurred while loading spec_helper.
> Failure/Error: require "rails/test_help"
> 
> NoMethodError:
>   undefined method `config' for nil
> # ./spec/spec_helper.rb:4:in `<top (required)>'
> No examples found.
> 
> 
> Finished in 0.00002 seconds (files took 0.63226 seconds to load)
> 0 examples, 0 failures, 1 error occurred outside of examples
> 
> /Users/jasl/.rbenv/versions/3.3.0/bin/ruby -I/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-support-3.13.1/lib:/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/lib /Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb failed
> ```

The error is [here](https://github.com/Uysim/pagy-cursor/pull/65/files#diff-7fc8afcf78366c15de1a1475a986fcdcd484e0a46440f12dc3771a7e0460f071R4).
In gemspec we just need version. But after this change we require the whole library. Which will cause error for the first bundle. 
For your case, you have install dependencies since before the code break. So you will be fine.

**Comment by jasl** [comment]
> > BTW, this branch works for my projects, but I don't know why `rake spec` has problem
> > ```
> > /Users/jasl/.rbenv/versions/3.3.0/bin/ruby -I/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-support-3.13.1/lib:/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/lib /Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
> > 
> > An error occurred while loading spec_helper.
> > Failure/Error: require "rails/test_help"
> > 
> > NoMethodError:
> >   undefined method `config' for nil
> > # ./spec/spec_helper.rb:4:in `<top (required)>'
> > No examples found.
> > 
> > 
> > Finished in 0.00002 seconds (files took 0.63226 seconds to load)
> > 0 examples, 0 failures, 1 error occurred outside of examples
> > 
> > /Users/jasl/.rbenv/versions/3.3.0/bin/ruby -I/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-support-3.13.1/lib:/Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/lib /Users/jasl/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/rspec-core-3.13.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb failed
> > ```
> 
> The error is [here](https://github.com/Uysim/pagy-cursor/pull/65/files#diff-7fc8afcf78366c15de1a1475a986fcdcd484e0a46440f12dc3771a7e0460f071R4). In gemspec we just need version. But after this change we require the whole library. Which will cause error for the first bundle. For your case, you have install dependencies since before the code break. So you will be fine.

I added back `version.rb`, but it still got the error. My project uses Ruby 3.3 with Rails 7.1 without issues.

**Comment by jasl** [comment]
interesting... only Ruby 3.3 with Rails 7.1 failed

**Pull request merged**